presto-main/src/test/java/io/prestosql/operator/scalar/TestArrayContainsSubarray.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import org.testng.annotations.Test;
+
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+
+public class TestArrayContainsSubarray
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testBasic()
+    {
+        assertFunction("contains_subarray(ARRAY [1,2,3,4,5,6], ARRAY[1,2])", BOOLEAN, true);
+        assertFunction("contains_subarray(ARRAY [1,2,3,4,5,6], ARRAY[1,2,10])", BOOLEAN, false);
+        assertFunction("contains_subarray(ARRAY [1,2,3,NULL,4,5,6], ARRAY[1,2,NULL])", BOOLEAN, true);
+        assertFunction("contains_subarray(ARRAY [1,2,3,4,5,6], ARRAY[1,2,NULL])", BOOLEAN, false);
+        assertFunction("contains_subarray(ARRAY ['1','2','3'], ARRAY['1','2'])", BOOLEAN, true);
+        assertFunction("contains_subarray(ARRAY ['1','2','3'], ARRAY['1','4'])", BOOLEAN, false);
+        assertFunction("contains_subarray(ARRAY [1.1,2.2,3.3], ARRAY[1.1,2.2])", BOOLEAN, true);
+        assertFunction("contains_subarray(ARRAY [1.1,2.2,3.3], ARRAY[1.1,2.2,5.5])", BOOLEAN, false);
```

Comment:
This should have cases that cover the middle of the array.  It should also verify the "contains all" vs the "contains sub-sequence" semantics.

Reply:
done